third_party/2and3/pymysql/cursors.pyi

```diff
@@ -22,12 +22,24 @@ class Cursor:
     def executemany(self, query: str, args) -> int: ...
     def callproc(self, procname, args=...): ...
     def fetchone(self) -> Optional[Gen]: ...
-    def fetchmany(self, size: Optional[int] = ...) -> Optional[Gen]: ...
+    def fetchmany(self, size: Optional[int] = ...) -> Union[Optional[Gen], List[Gen]]: ...
     def fetchall(self) -> Optional[Tuple[Gen, ...]]: ...
-    def scroll(self, value, mode=''): ...
+    def scroll(self, value: int, mode: str = ...): ...
     def __iter__(self): ...
 
 class DictCursor(Cursor):
     def fetchone(self) -> Optional[Dict[str, Any]]: ...
-    def fetchmany(self, size=None) -> Optional[Tuple[Dict[str, Any], ...]]: ...
+    def fetchmany(self, size: Optional[int] = ...) -> Optional[Tuple[Dict[str, Any], ...]]: ...
     def fetchall(self) -> Optional[Tuple[Dict[str, Any], ...]]: ...
+
+class DictCursorMixin:
+    dict_type = ...  # type: Any
+
+class SSCursor(Cursor):
+    def fetchall(self) -> Optional[Tuple[Gen, ...]]: ...
```

Comment:
Doesn't this return a list? https://github.com/PyMySQL/PyMySQL/blob/master/pymysql/cursors.py#L458

Reply:
I think you should give the correct type (`List[Gen]`) here, and `# type: ignore` the error. We should give the types as they actually are, even if they break LSP.